llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp

```diff
@@ -157,15 +157,17 @@ Instruction *emitCall(Module &M, Type *RetTy, StringRef FunctionName,
 // Insert instrumental annotation calls, that has no arguments (for example
 // work items start/finish/resume and barrier annotation.
 void insertSimpleInstrumentationCall(Module &M, StringRef Name,
-                                     Instruction *Position) {
+                                     Instruction *Position,
+                                     const DebugLoc &DL) {
   Type *VoidTy = Type::getVoidTy(M.getContext());
   ArrayRef<Value *> Args;
   Instruction *InstrumentationCall = emitCall(M, VoidTy, Name, Args, Position);
   assert(InstrumentationCall && "Instrumentation call creation failed");
+  InstrumentationCall->setDebugLoc(DL);
 }
 
 // Insert instrumental annotation calls for SPIR-V atomics.
-void insertAtomicInstrumentationCall(Module &M, StringRef Name,
+bool insertAtomicInstrumentationCall(Module &M, StringRef Name,
```

Comment:
I suggest leaving the function to be of a `void` type (changed in https://github.com/intel/llvm/pull/3691)

Reply:
I do not think that would be correct.  We have to set `IRModified` to `true` whenever we modify IR, so that we can clobber all analyses.  Since this function may exit without modifying IR, it makes sense to let it return `true/false`.